library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -440,12 +440,13 @@ private NetworkState checkSizeAllowedForNetwork(int networkType) {
      * create a {DownloadThread} and enqueue it into given
      * {@link Executor}.
      *
-     * @param collatedDownloadInfo
      * @return If actively downloading.
      */
     public boolean isReadyToDownload(CollatedDownloadInfo collatedDownloadInfo) {
         synchronized (this) {
-            return isClientReadyToDownload(collatedDownloadInfo) && isDownloadManagerReadyToDownload();
+            // This order MATTERS
+            // it means completed downloads will not be accounted for in later downloadInfo queries
+            return isDownloadManagerReadyToDownload() && isClientReadyToDownload(collatedDownloadInfo);
```

Comment:
it's `isDownloadManagerReadyToDownload` which checks if the download has already completed